src/Host/Client/Impl/Host/RemoteBrokerClient.cs

```diff
@@ -45,8 +50,12 @@ public RemoteBrokerClient(string name, Uri brokerUri, ICoreServices services, IC
                 ? new RHostDisconnectedException(Resources.Error_BrokerNotRunning, exception)
                 : await base.HandleHttpRequestExceptionAsync(exception);
         }
-        
+
         private bool ValidateCertificateHttpHandler(object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors) {
+            if(_cancellationToken.IsCancellationRequested) {
```

Comment:
Can you please try one more option?
`sender` should be of the `HttpWebRequest` type, and if it is, you should be able to check `HttpWebRequest.Status`